pkg/kvstore/allocator/allocator.go

```diff
@@ -169,7 +173,7 @@ func (k *kvstoreBackend) AllocateIDIfLocked(ctx context.Context, id idpool.ID, k
 
 // AcquireReference marks that this node is using this key->ID mapping in the kvstore.
 func (k *kvstoreBackend) AcquireReference(ctx context.Context, id idpool.ID, key allocator.AllocatorKey, lock kvstore.KVLocker) error {
-	keyString := key.GetKey()
+	keyString := string(encodeKey(key))
```

Comment:
This looks inefficient, the `encodeKey` could return a string because the `kvstore.Encode` returns a string.
```golang
func encodeKey(key allocator.AllocatorKey) []byte {
	return []byte(kvstore.Encode([]byte(key.GetKey())))
}
```
The even weirder part is that `etcdClient` encodes a `[]byte` into a `string`. So we are basically casting 4(?) times something that doesn't actually need any casting.
```golang
func (e *etcdClient) Encode(in []byte) string {
	return string(in)
}
```